grovedb/src/operations/get/query.rs

```diff
@@ -309,6 +319,94 @@ where {
         Ok((results, skipped)).wrap_with_cost(cost)
     }
 
+    /// Queries the backing store and returns element items by their value,
+    /// Sum Items are returned
+    pub fn query_item_value_or_sum(
+        &self,
+        path_query: &PathQuery,
+        allow_cache: bool,
+        decrease_limit_on_range_with_no_sub_elements: bool,
+        error_if_intermediate_path_tree_not_present: bool,
+        transaction: TransactionArg,
+    ) -> CostResult<(Vec<QueryItemOrSumReturnType>, u16), Error> {
+        let mut cost = OperationCost::default();
+
+        let (elements, skipped) = cost_return_on_error!(
+            &mut cost,
+            self.query_raw(
+                path_query,
+                allow_cache,
+                decrease_limit_on_range_with_no_sub_elements,
+                error_if_intermediate_path_tree_not_present,
+                QueryResultType::QueryElementResultType,
+                transaction
+            )
+        );
+
+        let results_wrapped = elements
+            .into_iterator()
+            .map(|result_item| match result_item {
+                QueryResultElement::ElementResultItem(element) => {
+                    match element {
+                        Element::Reference(reference_path, ..) => {
+                            match reference_path {
+                                ReferencePathType::AbsolutePathReference(absolute_path) => {
+                                    // While `map` on iterator is lazy, we should accumulate costs
+                                    // even if `collect` will
+                                    // end in `Err`, so we'll use
+                                    // external costs accumulator instead of
+                                    // returning costs from `map` call.
+                                    let maybe_item = self
+                                        .follow_reference(
+                                            absolute_path.as_slice().into(),
+                                            allow_cache,
+                                            transaction,
+                                        )
+                                        .unwrap_add_cost(&mut cost)?;
+
+                                    match maybe_item {
```

Comment:
i think we shall stop with `maybe_item` there and reuse the following code intended for non-references, because it's the same and reference will be already resolved

Reply:
I didn't understand this comment.